data_model/src/query/sorting.rs

```diff
@@ -38,13 +38,13 @@ impl Sorting {
     }
 }
 
-impl From<Sorting> for Vec<(&'static str, Name)> {
-    fn from(sorting: Sorting) -> Self {
-        if let Some(key) = sorting.sort_by_metadata_key {
-            return vec![(SORT_BY_KEY, key)];
-        }
-
-        Vec::new()
+impl Sorting {
+    /// Converts self to Vec of tuples to be used in queries
+    ///
+    /// The length of the output Vec is not constant and it's in (0..3)
+    pub fn into_query_parameters(self) -> Vec<(&'static str, Name)> {
```

Comment:
The length is actually either 0 or 1. Not sure if it makes sense to return `Vec<_>` here just for uniformity, when `Option<_>` suits better. However, it would require `builder.params` to accept this type as well.

Reply:
Changed to return impl IntiIterator uniformly. Under the hood Sorting has Option.